src/app/delegation_backend/src/delegation_backend/app_config.go

```diff
@@ -0,0 +1,98 @@
+package delegation_backend
+
+import (
+	"encoding/json"
+	"os"
+
+	logging "github.com/ipfs/go-log/v2"
+)
+
+func loadAwsCredentials(filename string, log logging.EventLogger) {
+	file, err := os.Open(filename)
+	if err != nil {
+		log.Errorf("Error loading credentials file: %s", err)
+		os.Exit(1)
+	}
+	defer file.Close()
+	decoder := json.NewDecoder(file)
+	var credentials AwsCredentials
+	err = decoder.Decode(&credentials)
+	if err != nil {
+		log.Errorf("Error loading credentials file: %s", err)
+		os.Exit(1)
+	}
+	os.Setenv("AWS_ACCESS_KEY_ID", credentials.AccessKeyId)
+	os.Setenv("AWS_SECRET_ACCESS_KEY", credentials.SecretAccessKey)
+}
+
+func LoadEnv(log logging.EventLogger) AppConfig {
+	var config AppConfig
+
+	configFile := os.Getenv("CONFIG_FILE")
+	if configFile != "" {
+		file, err := os.Open(configFile)
+		if err != nil {
+			log.Errorf("Error loading config file: %s", err)
+			os.Exit(1)
+		}
+		defer file.Close()
+		decoder := json.NewDecoder(file)
+		err = decoder.Decode(&config)
+		if err != nil {
+			log.Errorf("Error loading config file: %s", err)
+			os.Exit(1)
+		}
+	} else {
+		networkName := os.Getenv("CONFIG_NETWORK_NAME")
+		if networkName == "" {
+			log.Fatal("missing NETWORK_NAME environment variable")
```

Comment:
if networkName as well as others are not set we log fatal, but still proceeds with code execution. IMHO if there is an fatal entry in log it means that program should abort, while here we don't exit just  proceed with empty strings. Is it correct?

Reply:
Should be clarified knowing that `log.Fatalf` exits the execution.